java-frontend/src/main/java/org/sonar/java/resolve/JavaSymbol.java

```diff
@@ -678,12 +681,13 @@ public Boolean checkOverridingParameters(MethodJavaSymbol overridee, ClassJavaTy
         // Generics type should have same erasure see JLS8 8.4.2
 
         JavaType overrideeType = overridee.getParametersTypes().get(i);
-        if (classType.isParameterized()) {
+        overrideeType = resolve.typeSubstitutionSolver.applySiteSubstitutionToFormalParameters(Collections.singletonList(overrideeType), classType).get(0);
```

Comment:
I don't really like the fact that we have to inject the static field `resolve` from the `SemanticModel`, it looks like spaghetti code to me, adding to the mess.

Now, why not applying the substitution directly to all the parameters, before starting the loop?